### PR TITLE
Add koalas.concat

### LINKED ISSUE #231: koalas.concat
Similar to https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.concat.html#pandas.concat

This is basically a unionAll in Spark.

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR adds base of ks.concat. There are still many parameters to implement existent in Pandas'

Resolves #231

### CODE REVIEW

File: databricks/koalas/tests/test_dataframe.py [hunk 1]
```diff
@@ -489,3 +488,21 @@ def test_sample(self):
             kdf.sample()
         with self.assertRaises(NotImplementedError):
             kdf.sample(n=1)
+
+    def test_concat(self):
```
Comment (rxin) [inline databricks/koalas/tests/test_dataframe.py]: we should create a new file for the general utility functions defined in namespace.py

Comment (HyukjinKwon) [reply databricks/koalas/tests/test_dataframe.py]: okie

File: databricks/koalas/namespace.py [hunk 2]
```diff
@@ -538,6 +540,216 @@ def column_name(value):
     return kdf[remaining_columns]
 
 
+# TODO: there are many parameters to implement and support. See Pandas's pd.caoncat.
```
Comment (ueshin) [inline databricks/koalas/namespace.py]: nit: `pd.caoncat` -> `pd.concat`

File: databricks/koalas/namespace.py [hunk 3]
```diff
@@ -538,6 +540,216 @@ def column_name(value):
...
+def concat(objs, axis=0, join='outer', ignore_index=False):
```
Comment (ueshin) [inline databricks/koalas/namespace.py]: We don't need to add `axis`?

File: databricks/koalas/namespace.py [hunk 4]
```diff
@@ -538,6 +540,216 @@ def column_name(value):
...
+        ``Series`` is returned. When ``objs`` contains at least one
+        ``DataFrame``, a ``DataFrame`` is returned.
+
+    See Also
+    --------
+    DataFrame.merge
+
+    Notes
+    -----
+    The keys, levels, and names arguments are all optional.
```
Comment (ueshin) [inline databricks/koalas/namespace.py]: `optional` -> `not supported`? or we can remove the `Notes`?

File: databricks/koalas/namespace.py [hunk 5]
```diff
@@ -538,6 +540,216 @@ def column_name(value):
...
+    # DataFrame, Series ... & Series, Series ...
+    new_objs = []
+    for obj in objs:
+        if isinstance(obj, Series):
+            obj = obj.to_dataframe()
+        new_objs.append(obj)
+    objs = new_objs
+
+    # DataFrame, DataFrame, ...
+    indices_of_kdfs = list(map(lambda kdf: kdf._metadata.index_map, objs))
```
Comment (ueshin) [inline databricks/koalas/namespace.py]: just out of curiosity, is this the same as `[kdf._metadata.index_map for kdf in objs]`?

File: databricks/koalas/namespace.py [hunk 6]
```diff
@@ -538,6 +540,216 @@ def column_name(value):
...
+    index_of_first_kdf = indices_of_kdfs[0]
+    for index_of_kdf in indices_of_kdfs:
+        if index_of_first_kdf != index_of_kdf:
```
Comment (ueshin) [inline databricks/koalas/namespace.py]: If `ignore_index=True`, we can skip this check?

File: databricks/koalas/namespace.py [hunk 7]
```diff
@@ -538,6 +540,216 @@ def column_name(value):
...
+        concatenated = concatenated.unionByName(kdf._sdf)
+
+    if ignore_index:
+        result_kdf = DataFrame(concatenated.select(kdfs[0]._metadata.data_columns))
+    else:
+        result_kdf = DataFrame(concatenated, kdfs[0]._metadata.copy())
+
+    # Series, Series ...
+    if should_return_series:
+        return result_kdf[result_kdf.columns[0]]
```
Comment (ueshin) [inline databricks/koalas/namespace.py]: nit: `series._col` can be used?

File: databricks/koalas/namespace.py (hunk 5 above)
Comment (HyukjinKwon) [reply databricks/koalas/namespace.py]: oh, yes. that looks simpler.

File: databricks/koalas/namespace.py (hunk 3 above)
Comment (HyukjinKwon) [reply databricks/koalas/namespace.py]: Current behaviour is when axis is `0`. Let me throw an exception if the values are different.

**Comment by ueshin** [comment]
Thanks! merging to master.

**Pull request merged**